### ISSUE TITLE
Network Programming Team : Comment here

### ISSUE THREAD

**Issue by lordzuko** [OpenC-IIIT/prosfair#2]

**Comment by motoko-11** [comment]
The idea about using SSH gui forwarding to create the multiplayer environment has a flow.. what if the connection is disrupted... ?? @lordzuko It will hang the game in between for sometime.. and what if it creates some exception which makes the player got his session close.. has to restart the session... although it's possible for us to keep a autosave feature for per player basis to solve this problem..

**Comment by lordzuko** [comment]
autosave feature will be best thing to start with , but can we save the latest change in case the session closes ??

**Comment by motoko-11** [comment]
Can you be more specific @lordzuko

**Comment by lordzuko** [comment]
i mean in case the session closes, is the most recent state of the game is going to be saved in the local file , is there any guarantee ?

**Comment by motoko-11** [comment]
No ... but it should be ... so we should write down moves and states into a local file while sending it over network too.. with some sequence numbering.. that way the lowest common sequence number which both the users have is the last state from which both can start playing again.

**Comment by lordzuko** [comment]
why not we just save the current board state in .json file as we have started with ... that will have the current position of all the board pieces and we wont even need to do anything extra... what say ??

**Comment by motoko-11** [comment]
thats good suggestion.. still I think we will need both players to have knowledge of what the last state was of the other player... to get in sync..

**Comment by lordzuko** [comment]
what i suggested will have the position of both the players moves ... am i correct ?

**Comment by motoko-11** [comment]
yeah... I am not able to think of any case where it can fail... @lordzuko , If turns are strictly enforces, this can work just fine ..

**Comment by lordzuko** [comment]
:+1:
